### PR TITLE
Bench 122 add retired field to models

### PR DESCRIPTION
**Pull request by NogaAD**

### CODE REVIEW

File: answerking_app/utils/mixins/GenericMixins.py [hunk 1]
```diff
@@ -27,6 +27,15 @@ def update(self, request: Request, *args, **kwargs) -> Response:
             return item_exists_check(err)
 
 
+class RetireMixin(UpdateModelMixin):
+    def retire(self, request: Request, *args, **kwargs) -> Response:
+        try:
+            request.data['retired']: bool = True
+            return super().partial_update(request, *args, **kwargs)
+        except ObjectDoesNotExist as err:
+            return item_exists_check(err)
```
Comment (PietroConvalleAD) [inline answerking_app/utils/mixins/GenericMixins.py]: I should have pointed this out during the previous review but I didn't notice it, why do you do this check and not just return a 404?

Comment (NogaAD) [reply answerking_app/utils/mixins/GenericMixins.py]: not sure what do you mean. use get_or_404()?

Comment (PietroConvalleAD) [reply answerking_app/utils/mixins/GenericMixins.py]: yes, and even without the error should be handled by DRF anyway I believe

Comment (NogaAD) [reply answerking_app/utils/mixins/GenericMixins.py]: ok, done

**Review by MichaelOSullivanAnswer — APPROVED** [review]

File: answerking_app/tests/test_orderlines.py [hunk 2]
```diff
@@ -294,31 +294,3 @@ def test_delete_nonexistant_id_returns_not_found(self):
         # Assert
         self.assertEqual(expected, actual)
         self.assertEqual(response.status_code, 400)
-
```
Comment (JoeCSykes) [inline answerking_app/tests/test_orderlines.py:297]: could we put a test in making sure that when a retired item is in a order it is ignored please

Comment (NogaAD) [reply answerking_app/tests/test_orderlines.py:297]: I didn't do that as the other refactor order ticket will change things and I'll need to change the test again. I'll add test in the next ticket

Comment (JoeCSykes) [reply answerking_app/tests/test_orderlines.py:297]: ok 👍

**Comment by JoeCSykes** [comment]
could you delete the migration file and push again. It should pass the checks without it now. Then I will approve :)

**Review by JoeCSykes — APPROVED** [review]

**Pull request merged**